Comprehensive Aragon Bug Bounty Program (#107)
aragon/AGPs: AGPs/AGP-Comprehensive-Aragon-Bug-Bounty-Program.md

---
AGP: N/A
Title: Comprehensive Aragon Bug Bounty Program
Author: burrrata (@burrrata)
Status: Stage III
Track: Association
Created: 2019-10-02
---

# AGP-X: Comprehensive Aragon Bug Bounty Program

## Description of desired Association policy change

> This proposal is to create a comprehensive [Bug Bounty Program](https://wiki.aragon.org/dev/bug_bounty/) for all funded Aragon applications and tooling. 

Reviewer: ```suggestion
This proposal is to create a comprehensive [Bug Bounty Program](https://wiki.aragon.org/association/security/bug_bounty/) for all funded Aragon applications and tooling. 
```

Author: Thanks!

Author: Question: if I click `Resolve conversation` will it automatically merge your suggestions in?

Reviewer: @burrrata it appears not, due to the [issue](https://github.com/aragon/AGPs/pull/100#issuecomment-540270809) mentioned in the other thread. I will create a new PR so it's easier to make edits on these files later today.

### Bug Bounty Scope

> The scope of the [Aragon Association Bug Bounty Program](https://wiki.aragon.org/dev/bug_bounty/) will be expanded to make all Nest grantees and Flock team applications eligible. Applications will only be accepted into the bug bounty program if they have undergone a security audit, incorporated all major and critical security audit recommendations, and shipped to mainnet. 

Reviewer: ```suggestion
The scope of the [Aragon Association Bug Bounty Program](https://wiki.aragon.org/association/security/bug_bounty/) will be expanded to make all Nest grantees and Flock team applications eligible. Applications will only be accepted into the bug bounty program if they have undergone a security audit, incorporated all major and critical security audit recommendations, and shipped to mainnet. 
```

Author: Good catch. Thanks

> The scope of the [Aragon Association Bug Bounty Program](https://wiki.aragon.org/dev/bug_bounty/) will be expanded to make all Nest grantees and Flock team applications eligible. Applications will only be accepted into the bug bounty program if they have undergone a security audit, incorporated all major and critical security audit recommendations, and shipped to mainnet. 

Reviewer 2: We will need to increase the scope to include the Aragon Court soon, I would preemptively add it to the list even though the program won't launch until A1 considers de codebase ready for it.

Author: I believe that this would be covered as A1, a Flock team, is building the Aragon Court. Added a comment mentioning the Aragon Court to make it explicitly clear that it would be covered.

### Bug Bounty Levels

> The [Aragon Association Bug Bounty Program](https://wiki.aragon.org/dev/bug_bounty/) would increase the Bug Bounty rewards. 

Reviewer: ```suggestion
The [Aragon Association Bug Bounty Program](https://wiki.aragon.org/association/security/bug_bounty/) would increase the Bug Bounty rewards. 
```

Author: Good catch. Thanks

Currently the Aragon Association Bug Bounty program covers:
- Critical (CVSS 9.0 - 10.0): $5,000 - $50,000
- Major (CVSS 7.0 - 8.9): $2,500 - $5,000
- Medium (CVSS 4.0 - 6.0): $1,000 - $2,500
- Low (CVSS 1.0 - 3.9): $500 - $1,000

> If we are spending millions of dollars on Flock teams and hundreds of thousands of dollars on Nest projects, we can at least spend a little more on our bug bounties. Aragon DAOs currently hold hundreds of thousands of dollars in thier vaults, and soon that will be millions. Paying out $10,000 for a critical vulnerability is simply insufficient. 

Reviewer: ```suggestion
If we are spending millions of dollars on Flock teams and hundreds of thousands of dollars on Nest projects, we can at least spend a little more on our bug bounties. Aragon DAOs currently hold hundreds of thousands of dollars in their vaults, and soon that will be millions. Paying out $10,000 for a critical vulnerability is simply insufficient.
```

Author: Good catch. Thanks

> The Aragon Association Bug Bounty program will be upgraded to the following levels:

Reviewer 2: I would add a cap of how much money the bounty pool has per year, so we cap our liabilities. I think 1m a year would be a reasonable amount for the stage that we are at.

Author: Good idea! Added this in as well

- Critical (CVSS 9.0 - 10.0): $50,000 - $150,000
- Major (CVSS 7.0 - 8.9): $25,000 - $50,000
- Medium (CVSS 4.0 - 6.0): $10,000 - $25,000
- Low (CVSS 1.0 - 3.9): $5000 - $10,000

> Considering that these bounties will only be paid out in the case that there is a problem, it's a small price to pay to minimize mission critical errors and build trust in Aragon as a platform.

Reviewer: ```suggestion
Considering that these bounties will only be paid out in the case that there is a problem, it's a small price to pay to minimize mission-critical errors and build trust in Aragon as a platform.
```

Author: Good catch. Thanks

## Motivation for changing this Association policy

Aragon is now shipping more apps through [Autark](https://www.autark.xyz/), [Aragon Black](https://aragon.black/), and the [Nest program](https://github.com/aragon/nest/). These apps are amazing and could unlock tons of value, but only if people use them…

> People are wary of DAOs because of [“the DAO” hack](http://hackingdistributed.com/2016/05/27/dao-call-for-moratorium/). For people to trust (and thus use) Aragon we need to go above and beyond to prove that the Aragon platform, and all major releases of Aragon apps, are secure. This is easier said than done.

Reviewer 3: The DAO (capital T)

Author: Good catch. Thanks

- Security audits are not perfect. Even with an audit all you know is what was reported. They might have missed something.

> - Security audits are highly technical and the process is opaque to people who are not involved in the Ethereum security commuinty.

Reviewer: ```suggestion
- Security audits are highly technical and the process is opaque to people who are not involved in the Ethereum security community.
```

Author: Good catch. Thanks

- Security audits are expensive. Having strategic and financial help to navigate that negotiation is extremely important!

> Aragon is trying to attract talent and users. Having the worlds easiest to build on _and_ most secure platform is a huge selling point. If developers see that they will have help shipping professional and production ready applications they are more likely to choose to build on Aragon vs other platforms. If users trust that all major Aragon apps are secure, they’re more likely to use them. To do this we need a multi layered approach to security. This can include audits for all major projects (Nest and Flock) as well as a comprehensive Bug Bounty program that covers all major apps. This is a small price to pay to establish credibility and trust in the Aragon platform and developer ecosystem.

Reviewer: ```suggestion
Aragon is trying to attract talent and users. Having the world's easiest to build on _and_ most secure platform is a huge selling point. If developers see that they will have help shipping professional and production-ready applications they are more likely to choose to build on Aragon vs other platforms. If users trust that all major Aragon apps are secure, they’re more likely to use them. To do this we need a multi-layered approach to security. This can include audits for all major projects (Nest and Flock) as well as a comprehensive Bug Bounty program that covers all major apps. This is a small price to pay to establish credibility and trust in the Aragon platform and developer ecosystem.
```

Author: Good catch. Thanks
